lib/cfn-nag/custom_rules/AmazonMQBrokerUserPasswordRule.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require 'cfn-nag/util/enforce_reference_parameter'
+require 'cfn-nag/util/enforce_string_or_dynamic_reference'
+require_relative 'base'
+
+class AmazonMQBrokerUserPasswordRule < BaseRule
+
+  def rule_text
+    'Amazon MQ Broker resource Users property should exist and its Password property value ' +
+    'should not show password in plain text, resolve an unsecure ssm string, or have a default value for parameter.'
+  end
+
+  def rule_type
+    Violation::FAILING_VIOLATION
+  end
+
+  def rule_id
+    'F52'
+  end
+
+  def check_password_property(cfn_model, user)
```

Comment:
'check' is a bit opaque as a term.  while i would never say comments are never necessary, imho writing them usually means something isn't obvious from the code itself.

user_has_insecure_password? is probably a better name